app/mailers/references_header_builder.rb

```diff
@@ -0,0 +1,52 @@
+# Builds RFC 5322 compliant References headers for email threading
+#
+# This module provides functionality to construct proper References headers
+# that maintain email conversation threading according to RFC 5322 standards.
+module ReferencesHeaderBuilder
+  # Builds a complete References header for an email reply
+  #
+  # According to RFC 5322, the References header should contain:
+  # - All message IDs from the conversation thread in chronological order
```

Comment:
@scmmishra Have you tested the case when the email is partially available in Chatwoot and we don't have the complete thread? How does this behave in such cases?